File: app/src/main/java/com/luigivampa92/xlogger/DebugLastLogStorage.java
package com.luigivampa92.xlogger;

import android.content.Context;
import android.content.SharedPreferences;

public class DebugLastLogStorage {

    private SharedPreferences sharedPreferences;

    public DebugLastLogStorage(Context context) {
        sharedPreferences = context.getSharedPreferences("DEBUG", Context.MODE_PRIVATE);
    }

    public void saveLastLog(String logText) {
        sharedPreferences.edit()
                .putString("LAST_LOG", logText)
                .apply();
    }

    public String getLastLog() {
        return sharedPreferences.getString("LAST_LOG", "");
    }
}

File: app/src/main/java/com/luigivampa92/xlogger/data/DebugLastLogStorage.kt
package com.luigivampa92.xlogger.data

import android.content.Context
import android.content.SharedPreferences

class DebugLastLogStorage(context: Context) {

    private val sharedPreferences: SharedPreferences

    init {
        sharedPreferences = context.getSharedPreferences("DEBUG", Context.MODE_PRIVATE)
    }

    fun saveLastLog(logText: String?) {
        sharedPreferences.edit()
            .putString("LAST_LOG", logText)
            .apply()
    }

    fun getLastLog(): String? {
        return sharedPreferences.getString("LAST_LOG", "")
    }
}